templates/platform_landing_zone/modules/management_resources/terraform.tf

```diff
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     azurerm = {
       source  = "hashicorp/azurerm"
-      version = "~> 3.107"
+      version = "~> 4.0"
```

Comment:
This module only calls other modules and as such should not need a requried_providers entry - it's only minor so won't hurt to leave it in.